docs/source/notes/randomness.rst

```diff
@@ -0,0 +1,41 @@
+
+Reproducibility
+===============
+
+Completely reproducible results are not guaranteed across PyTorch releases,
+individual commits or different platforms. Furthermore, results need to be
+reproducible between CPU and GPU executions, even when using identical seeds.
+
+However, in order to make computations deterministic on your specific problem on
+one specific platform and PyTorch release, there are a couple of steps to take.
+
+There are several pseudorandom number generators involved in various layers of
+the library, which you will need to seed manually to make runs reproducible.
+
+Numpy
+.....
+The numpy rng can be seeded with::
+
+    import numpy as np
+    np.random.seed(0)
+
+PyTorch
+.......
+When running on the CPU, you should seed the RNG with::
+
+    import torch
+    torch.manual_seed(0)
+
+While this will seed all CUDA RNGs as well, you can do so explicitly with
+:meth:`torch.cuda.manual_seed` for the current device, or
+:meth:`torch.cuda.manual_seed_all` to seed the RNG for all GPUs
```

Comment:
It's worth noting why you might *not* want to `manual_seed_all`, which is that the time it takes scales with the number of GPUs you seed, so if you have a lot of GPUs but are only using one of them, `manual_seed_all` will needlessly bloat your load time,

Reply:
`torch.manual_seed` will set all devices lazily, including all CUDA GPUs.
It USED to be that the behavior of `torch.manual_seed` only seeded CPU.
Hence, this section can simply be said: `To seed all devices with a fixed RNG, set `torch.manual_seed(0)`.

In fact, @ezyang wrote that patch :)